lib/src/rules/curly_braces_in_flow_control_structures.dart

```diff
@@ -86,17 +86,22 @@ class _Visitor extends SimpleAstVisitor {
 
   @override
   void visitIfStatement(IfStatement node) {
-    if (node.elseStatement == null) {
+    var elseStatement = node.elseStatement;
+    var parent = node.parent;
+    if (elseStatement == null &&
+        !(parent is IfStatement && parent.elseStatement == node)) {
```

Comment:
I think this condition can (and maybe should) be removed. I believe it was attempting to guard against a case that can no longer occur.